components/script/dom/bindings/root.rs

```diff
@@ -291,6 +291,25 @@ impl<T: DomObject> MutNullableDom<T> {
         self.set(None);
         value
     }
+
+    /// Returns the nullable status of this object.
+    pub(crate) fn is_null(&self) -> bool {
+        unsafe { (*self.ptr.get()).is_none() }
+    }
+
+    /// Runs the given callback on the object if it's not null.
+    pub(crate) fn if_is_some<F>(&self, cb: F) -> Option<&MutNullableDom<T>>
+    where
+        F: FnOnce(&T) -> &MutNullableDom<T>,
```

Comment:
Why is `F` restrained to returning a `&MutNullableDom<T>`? Can the return type be generic too?